src/curl.cpp

```diff
@@ -1920,7 +1920,7 @@ bool S3fsCurl::CreateCurlHandle(bool only_pool, bool remake)
     AutoLock lock(&S3fsCurl::curl_handles_lock);
 
     if(hCurl && remake){
-        if(!DestroyCurlHandle(false)){
+        if(!DestroyCurlHandle(false, true, true)){
```

Comment:
Should this pass `AutoLock::ALREADY_LOCKED` instead?

Reply:
It has changed.